### ISSUE TITLE
using this wrapper in MVC

### ISSUE THREAD

**Issue by niroam** [mfilippov/vimeo-dot-net#13]
Hi,

any idea why using this wrapper in an MVC asp web project causes it to hang once the Autherization is complete and any function is called (i.e client.GetAccountInformation(); )

i can see it open a socket to the API url but seems to hang after that. 

i have downloaded your project and run the tests which work fine when i add my access token .. but am stuck why it hangs when its inside a MVC web project

Cheers
Niro

**Comment by renattomachado** [comment]
My problema is in method GetVideos() in MVC application. I investigated and the error return in response in request GetVideos() is 
> {"Error retrieving account videos.\r\nCode: BadRequest\r\nMessage: {\"error\":\"You must provide a search query.\"}"}

I'm tried resolve this problem. How my time is small, I should resolve yet in this week.

**Comment by KeeneState** [comment]
Struggled with this myself using MVC. I believe this is the same issue as [issue 8](https://github.com/scommisso/vimeo-dot-net/issues/8). It should go away with using the async variation of the method causing the program to hang.

For example, in .net 4.5+:

    VimeoClient client = CreateAuthenticatedClient();
    User account = client.GetAccountInformation();
    
could be rewritten as:

    private async Task<User> GetUser(VimeoClient client){
        User user = await client.GetAccountInformationAsync();
        return user;
    }
    ...
    VimeoClient client = CreateAuthenticatedClient()
    Task<User> account = GetUser(client);

**Comment by scommisso** [comment]
I posted this message on another issue - I'm looking for someone who wants to take over this repo. Any volunteers?

**Comment by scommisso** [comment]
#43 should fix this behavior
